crates/sui-framework/sources/governance/staking_pool.move

```diff
@@ -352,6 +352,11 @@ module sui::staking_pool {
             if (total_sui_withdraw_amount >= principal_withdraw_amount)
                 total_sui_withdraw_amount - principal_withdraw_amount
             else 0;
+        // This may happen when we are withdrawing everything from the pool and
+        // the rewards pool balance may be 1 less than reward_withdraw_amount.
```

Comment:
Might make sense to assert that the leftover amount is 1 to check this assumption.